code/modules/mob/living/carbon/human/human_update_icons.dm

```diff
@@ -573,58 +573,99 @@ There are several things that need to be remembered:
 
 	return icon(female_clothing_icon)
 
-// These coordonates point to roughly somewhere in the middle of the left leg
-// Used in approximating what color the pants of clothing should be
+/// Modifies a sprite to conform to digitigrade body shapes
+/proc/wear_digi_version(icon/base_icon, obj/item/item, key, greyscale_colors)
+	ASSERT(istype(item), "wear_digi_version: no item passed")
+	ASSERT(istext(key), "wear_digi_version: no key passed")
+	if(isnull(greyscale_colors) || length(SSgreyscale.ParseColorString(greyscale_colors)) > 1)
+		greyscale_colors = item.get_general_color(base_icon)
+
+	var/index = "[key]-[item.type]-[greyscale_colors]"
+	var/static/list/digitigrade_clothing_cache = list()
+	var/icon/resulting_icon = digitigrade_clothing_cache[index]
+	if(!resulting_icon)
+		var/digi_icon_state
+		if(istype(item, /obj/item/clothing/under))
+			digi_icon_state = "jumpsuit_worn"
+		else if(istype(item, /obj/item/clothing/suit))
+			digi_icon_state = "oversuit_worn"
+		else if(istype(item, /obj/item/clothing/shoes/sneakers))
+			digi_icon_state = "sneakers_worn"
+		else if(istype(item, /obj/item/clothing/shoes))
+			digi_icon_state = "boots_worn"
```

Comment:
I'm not sure why you got rid of the `generate_digitigrate_icons` proc, it's a lot cleaner solution than needing a big istype to add icon states